preload/main.c

```diff
@@ -22,28 +23,40 @@ int ioctl (int d, int request, char *argp) {
     if (orig_ioctl == NULL) {
         orig_ioctl = dlsym(RTLD_NEXT, "ioctl");
     }
+
+    /* We only care for TIOCGWINSZ ioctls */
+    if (request != 0x5413) {
+        return orig_ioctl(d, request, argp);
+    }
+
     static int max_rows = -1;
+
+    /* ioctl gets called once before the perl module had the time to determine
+     * the right size! Leave max_rows negative to indicat that it still needs to
+     * be read from the SHELLEX_SIZE_FILE */
+
     if (max_rows < 0 ) {
-        char *str = getenv("SHELLEX_MAX_ROWS");
+
+      char *fname = getenv("SHELLEX_SIZE_FILE");
```

Comment:
As annoying as it is to need to care about such BS, but the indentation here is wrong.